eZ/Publish/Core/FieldType/ImageAsset/Type.php

```diff
@@ -0,0 +1,279 @@
+    {
+        return new Value();
+    }
+
+    /**
+     * Returns if the given $value is considered empty by the field type.
+     *
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function isEmptyValue(SPIValue $value): bool
+    {
+        return null === $value->destinationContentId;
+    }
+
+    /**
+     * Inspects given $inputValue and potentially converts it into a dedicated value object.
+     *
+     * @param int|string|\eZ\Publish\API\Repository\Values\Content\ContentInfo|\eZ\Publish\Core\FieldType\Relation\Value $inputValue
+     *
+     * @return \eZ\Publish\Core\FieldType\ImageAsset\Value The potentially converted and structurally plausible value.
+     */
+    protected function createValueFromInput($inputValue): Value
+    {
+        if ($inputValue instanceof ContentInfo) {
+            $inputValue = new Value($inputValue->id);
+        } elseif (is_int($inputValue) || is_string($inputValue)) {
+            $inputValue = new Value($inputValue);
+        }
+
+        return $inputValue;
+    }
+
+    /**
+     * Throws an exception if value structure is not of expected format.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException If the value does not match the expected structure.
+     *
+     * @param \eZ\Publish\Core\FieldType\ImageAsset\Value $value
+     */
+    protected function checkValueStructure(BaseValue $value): void
+    {
+        if (!is_int($value->destinationContentId) && !is_string($value->destinationContentId)) {
+            throw new InvalidArgumentType(
+                '$value->destinationContentId',
+                'string|int',
+                $value->destinationContentId
+            );
+        }
+    }
+
+    /**
+     * Returns information for FieldValue->$sortKey relevant to the field type.
+     * For this FieldType, the related object's name is returned.
+     *
+     * @param \eZ\Publish\Core\FieldType\Relation\Value $value
+     *
+     * @return bool
+     */
+    protected function getSortInfo(BaseValue $value): bool
+    {
+        return false;
+    }
+
+    /**
+     * Converts an $hash to the Value defined by the field type.
+     *
+     * @param mixed $hash
+     *
+     * @return \eZ\Publish\Core\FieldType\ImageAsset\Value $value
+     */
+    public function fromHash($hash): Value
+    {
+        if ($hash) {
+            return new Value($hash['destinationContentId']);
+        }
+
+        return new Value();
+    }
+
+    /**
+     * Converts a $Value to a hash.
+     *
+     * @param \eZ\Publish\Core\FieldType\ImageAsset\Value $value
+     *
+     * @return array
+     */
+    public function toHash(SPIValue $value): array
+    {
+        return [
+            'destinationContentId' => $value->destinationContentId,
+        ];
+    }
+
+    /**
+     * Returns whether the field type is searchable.
+     *
+     * @return bool
+     */
+    public function isSearchable(): bool
```

Comment:
This method is same as in parent class, so it is redundant

Reply:
I agree with @mikadamczyk

> I'm sorry to disagree in this case. Relying on the fact that the current implementation of FieldType base core class returns false is risky. 

Not in this case: `\eZ\Publish\Core\FieldType\FieldType` is the abstract class . If I cannot rely on methods inherited from it, then I should implement `\eZ\Publish\SPI\FieldType\FieldType` from scratch.